src/dnm.rs

```diff
@@ -199,7 +200,18 @@ impl <'dnmrange> DNMRange <'dnmrange> {
         trimmed_start = self.start;
         trimmed_end = self.start;
     }
-    DNMRange {start : trimmed_start, end: trimmed_end, dnm: self.dnm}
+
+    //decrease the byte index to the next non-boundary index
```

Comment:
I think this should be a part of the main loop rather than a final sanity check, then we can avoid some edge cases. Compound unicode characters are a tricky thing in the first place, but the `is_char_boundary` check goes a long way to stay safe.